float32[] x_position
float32[] y_position
float32[] x_velocity
float32[] y_velocity
float32[] velocity_magnitude
